client-js/app/elements/labrad-nodes.ts

```diff
@@ -253,29 +265,72 @@ export class LabradNodes extends polymer.Base {
   @property({type: Number, value: 0})
   kick: number;
 
+  @property({type: Boolean, value: false, notify: true})
+  isAutostartFiltered: boolean;
+
+  @property({type: Object, value: []})
+  globalServers: ServerInfo[];
+
+  @property({type: Object, value: []})
+  localServers: ServerInfo[];
+
+  @property({type: Object, value: []})
+  globalServersFiltered: ServerInfo[];
+
+  @property({type: Object, value: []})
+  localServersFiltered: ServerInfo[];
+
   private lifetime = new Lifetime();
-  // Used to manage cleanup operations that should be performed when this
-  // component is destroyed. We register all such cleanup operations (such as
-  // Observable callback removals) with this lifetime, and then close it when
-  // the component is detached.
 
-  /**
-   * Polymer lifecyle callback invoked when this component is detached from the
-   * DOM and ready to be cleaned up.
-   */
+  @listen('labrad-instance-controller::ready')
+  onLabradInstanceControllerReady(event) {
+    // If a new labrad-instance-controller has readied, we need to broadcast
+    // the state of each controller of that server type to all others of the
+    // same type. This informs existing servers of the new server state, and
+    // informs the new server of the existing servers' states.
+    const instances = Polymer.dom(this.root).querySelectorAll('labrad-instance-controller');
+    for (var i = 0; i < instances.length; i++) {
```

Comment:
You can use `const` with the `for..of` syntax.

```TypeScript
for (const instance of instances) {
  ...
}
```
I have done this, but because of the cast, it's not exactly nicer..